### PR TITLE
feat: prepare for GenAI class

### PR DESCRIPTION
**Pull request by donlelef**
Small updates, including:
- switching from pdm to poetry
- fixing a small mistake in notebook 3
- resolving deprecation warning
- updating docs

### CODE REVIEW

File: src/movie_buddy/utils.py [hunk 1]
```diff
@@ -19,7 +20,7 @@ def add_sentences(data: pd.DataFrame, sentences: list[str], *, encoder) -> pd.Da
     return add_umap_to_dataset(all_sentences, reduced_encodings)
 
 
-def plot_movies(data: pd.DataFrame, *, sample: int = 5000) -> None:
```
Comment (baggiponte) [inline src/movie_buddy/utils.py:22]: We realised a sampling parameter would make the viz faster, but if you noticed you don't need it that's ok to remove.

File: pyproject.toml [hunk 2]
```diff
@@ -1,36 +1,28 @@
...
-dev = [
-    "pre-commit>=3.6.2",
-    "nbstripout>=0.7.1",
-]
-ide = [
-    "jupyterlab>=4.1.2",
-    "catppuccin-jupyterlab>=0.2.0"
-]
+[tool.poetry.dependencies]
+python = "^3.10"
```
Comment (baggiponte) [inline pyproject.toml:12]: maybe you want to pin this at `==3.10` to avoid clashes with colab?

**Review by baggiponte — APPROVED** [review]
Ok to me! Why did you prefer to add the if check on whether the collection exists, rather than using `recreate_collection`?

**Comment by donlelef** [comment]
> Ok to me! Why did you prefer to add the if check on whether the collection exists, rather than using `recreate_collection`?

This is just because recreate_collection is now deprecated by qdrant.

File: src/movie_buddy/utils.py (hunk 1 above)
Comment (donlelef) [reply src/movie_buddy/utils.py:22]: Yes, but it was not used in the function :)

File: pyproject.toml (hunk 2 above)
Comment (donlelef) [reply pyproject.toml:12]: Good point.

**Pull request merged**